dali/operators/generic/transpose/transpose.h

```diff
@@ -48,34 +48,53 @@ template <typename ShapeT = int>
 void PrepareArguments(SmallVector<ShapeT, kStaticShapeElements> &shape, VecInt &perm,
                       bool transpose = false) {
   DALI_ENFORCE(shape.size() == perm.size());
+  const int N = shape.size();
+
+  // This will be oterwise reduced to empty shape, and we still want to have some
+  // notion of non-empty shape left
+  if (volume(shape) == 1) {
```

Comment:
`transpose` aside, the code below seems to do the same in linear time and is (arguably) easier to understand:
```c++
  if (volume(shape) == 1) {
    shape = {1};
    perm = {0};
    return;
  }

  SmallVector<int, kStatiShapeElements> coord_map, tmp_perm;
  SmallVector<ShapeT, kStatiShapeElements> tmp_shape;
  int N = shape.size();
  coord_map.resize(N);
  
  for (int i = 0, skip = 0; i < N; i++) {
    if (shape[i] == 1)
      skip++;
    else
      tmp_shape.push_back(shape[i]);
    coord_map[i] = i - skip;
  }

  for (int i = 0; i < N; i++) {
    if (shape[perm[i]] == 1) continue;
    tmp_perm.push_back(coord_map[perm[i]]);
  }

  perm = std::move(tmp_perm);
  shape = std::move(tmp_shape);
```

Reply:
:+1: 
adjusted a bit, and added some comments.